File: examples/cdk-annotations/cdk-ts/storage-stack.ts
import { Stack, StackProps, Duration, RemovalPolicy } from "aws-cdk-lib";
import { Construct } from "constructs";
import * as s3 from "aws-cdk-lib/aws-s3";
import * as dynamodb from "aws-cdk-lib/aws-dynamodb";

/* @rs:config sheet: ストレージ基盤 (TS) */

/* @rs:category ストレージ */
export class StorageStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    /* @rs:category S3 */
    new s3.Bucket(this, "DataBucket", {
      bucketName: "myapp-data-prod", // @rs バケット名 @rs:remarks 命名規則 <app>-data-<env>
      versioned: true, // @rs バージョニング @rs:default false
      removalPolicy: RemovalPolicy.RETAIN, // @rs 削除ポリシー @rs:remarks 本番は RETAIN 必須
      encryption: s3.BucketEncryption.S3_MANAGED, // @rs 暗号化方式
    });

    /* @rs:category DynamoDB */
    new dynamodb.Table(this, "SessionsTable", {
      tableName: "sessions", // @rs テーブル名
      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST, // @rs 課金モード @rs:default PROVISIONED
      timeToLiveAttribute: "expiresAt", // @rs TTL属性 @rs:remarks 期限切れセッションを自動削除
      pointInTimeRecovery: true, // @rs PITR @rs:default false @rs:remarks 本番では有効
    });
  }
}

File: examples/cdk-annotations/cdk-py/storage_stack.py
from aws_cdk import Stack, Duration, RemovalPolicy
from aws_cdk import aws_s3 as s3
from constructs import Construct

# @rs:config sheet: ストレージ基盤 (Py)

# @rs:category アプリ設定
MAX_CONNECTIONS = 100      # @rs 最大接続数 @rs:default 50
TIMEOUT_SECONDS = 30       # @rs タイムアウト秒 @rs:remarks SLA は 3 秒以内
LOG_LEVEL = "INFO"         # @rs ログレベル @rs:default WARN

# @rs:category S3
BUCKET_NAME = "myapp-data-prod"   # @rs バケット名 @rs:remarks 命名規則 <app>-data-<env>
VERSIONED = True                  # @rs バージョニング @rs:default False

# @rs:category キャパシティ
settings = {
    "read_capacity": 25,          # @rs 読み込みキャパシティ @rs:default 5
    "removal_policy": "RETAIN",   # @rs 削除ポリシー @rs:remarks 本番は RETAIN 必須
}
